agent/engine/task_manager.go

```diff
@@ -277,6 +289,21 @@ func (mtask *managedTask) waitSteady() {
 	}
 }
 
+// getWaitSteadyCheckCancelFunc returns the cancel func for the steadyStateCheck wait
+func (mtask *managedTask) getWaitSteadyCheckCancelFunc() context.CancelFunc {
```

Comment:
rather than surfacing the cancel func like this to an entity that's external to the task, i'd rather have a listening channel and trigger a cancel based on an event being received from that channel

Reply:
yes, that would be a better approach. But this part of code is only useful in the test, having a listening channel will add some complexity into the code(we will need a delicated goroutine listener or reuse the waitEvent) and I don't see a use case where we need this beyond the test. What's your opinion?

Reply:
The current approach feels hacky even for testing IMO. Another alternative is to implement a new `cancelSteadyStateWait()` method and wrap all of this logic within that. Also, for all new members (fields and methods) that you added specifically for testing, please make sure that you add documentation stating the same